src/microbe_stage/MicrobeCamera.cs

```diff
@@ -223,13 +192,15 @@ public override void _Process(double delta)
         // ReSharper disable once CompareOfFloatsByEqualityOperator
         if (lastSetLightLevel != lightLevel)
         {
-            UpdateLightLevel((float)delta);
+            backgroundPlane.UpdateLightLevel((float)delta);
         }
 
         if (AutoProcessWhilePaused && PauseManager.Instance.Paused)
         {
             UpdateCameraPosition(delta, null);
         }
+
+        backgroundPlane.SetWorldPosition(new Vector2(Position.X, Position.Z));
```

Comment:
Shouldn't the background plane handle this or? (so it can just look at its own position to update the shader parameters)

Reply:
Apparently the background plane's world position doesn't change when it's in a subviewport, so the camera has to handle that

Reply:
Actually, I've just realized that the position of the `BackgroundPlane` does change, just not relative to the camera. So, `GlobalPosition` has to be used instead of `Position`. Otherwise, the implementation that you've described is in fact what already is done.

Reply:
Ah of course. I didn't notice you had made that mistake. It took me a while but I've almost entirely stopped making the mistake of using Position / WorldPosition when I should have used the other one.